jax/_src/numpy/polynomials/chebyshev.py

```diff
@@ -0,0 +1,605 @@
+    'chebvander', 'chebfit', 'chebtrim', 'chebroots', 'chebpts1',
+    'chebpts2', 'chebval2d', 'chebval3d', 'chebgrid2d',
+    'chebgrid3d', 'chebvander2d', 'chebvander3d', 'chebcompanion',
+    'chebgauss', 'chebweight', 'chebinterpolate'
+]
+
+#Not compatible with JIT
+chebtrim = pu.trimcoef
+
+@jit
+def _cseries_to_zseries(c):
+    n = c.size
+    zs = jnp.zeros(2*n-1, dtype=c.dtype)
+    zs = zs.at[n-1:].set(c/2)
+    return zs + zs[::-1]
+
+@jit
+def _zseries_to_cseries(zs):
+    n = (zs.size + 1)//2
+    c = zs[n-1:].copy()
+    c = c.at[1:n].multiply(2)
+    return c
+
+@jit
+def _zseries_mul(z1, z2):
+    return jnp.convolve(z1, z2)
+
+@jit
+def _zseries_div(z1, z2):
+    z1 = z1.copy()
+    z2 = z2.copy()
+    lc1 = len(z1)
+    lc2 = len(z2)
+
+    if lc2 == 1:
+        z1 /= z2
+        return z1, z1[:1]*0
+    elif lc1 < lc2:
+        return z1[:1]*0, z1
+    else:
+        dlen = lc1 - lc2
+        scl = z2[0]
+        z2 /= scl
+        quo = jnp.empty(dlen + 1, dtype=z1.dtype)
+        i = 0
+        j = dlen
+
+        while i < j:
+            r = z1[i]
+            quo = quo.at[i].set(z1[j])
+            quo = quo.at[dlen - i].set(r)
+            tmp = r*z2
+            z1 = z1.at[i:i+lc2].add(-tmp)
+            z1 = z1.at[j:j+lc2].add(-tmp)
+            i += 1
+            j -= 1
+        r = z1[i]
+        quo = quo.at[i].set(r)
+        tmp = r*z2
+        z1 = z1.at[i:i+lc2].add(-tmp)
+        quo /= scl
+        rem = z1[i+1:i-1+lc2].copy()
+        return quo, rem
+
+@jit
+def _zseries_der(zs):
+    n = len(zs)//2
+    ns = jnp.array([-1, 0, 1], dtype=zs.dtype)
+    zs *= jnp.arange(-n, n+1)*2
+    d, r = _zseries_div(zs, ns)
+    return d
+
+@jit
+def _zseries_int(zs):
+    n = 1 + len(zs)//2
+    ns = jnp.array([-1, 0, 1], dtype=zs.dtype)
+    zs = _zseries_mul(zs, ns)
+    div = jnp.arange(-n, n+1)*2
+    zs = zs.at[:n].divide(div[:n])
+    zs = zs.at[n+1:].divide(div[n+1:])
+    zs = zs.at[n].set(0)
+    return zs
+
+@jit
+def poly2cheb(pol):
+    [pol] = pu.as_series([pol])
+    deg = len(pol) - 1
+    res = 0
+    for i in range(deg, -1, -1):
+        res = chebadd(chebmulx(res), pol[i])
+    return res[:-1]
+
+@jit
+def cheb2poly(c):
+    [c] = pu.as_series([c])
+    n = len(c)
+    if n < 3:
+        return c
+    else:
+        c0 = jnp.asarray(c[-2])
+        c1 = jnp.asarray(c[-1])
+
+        for i in range(n-1, 1, -1):
+            tmp = c0
+            c0 = chebsub(c[i-2], c1)
+            if len(c1.shape) == 0:
+                prd = jnp.empty(2, dtype=c1.dtype)
+            else:
+                prd = jnp.empty(len(c1)+1, dtype=c1.dtype)
+            prd = prd.at[0].set(0)
+            prd = prd.at[1:].set(c1)
+            c1 = chebadd(tmp, prd*2)
+        prd = jnp.empty(len(c1)+1, dtype=c1.dtype)
+        prd = prd.at[0].set(0)
+        prd = prd.at[1:].set(c1)
+        return chebadd(c0, prd)
+
+#Chebyshev default domain
+chebdomain = jnp.array([-1, 1])
```

Comment:
We cannot create jax.numpy arrays at import time, because in some systems XLA will not yet be initialized when this is executed. Options are (1) use `np.array`, or (2) if `jnp.array` is important, define this lazily using a module `__getattr__`